lisa/nic.py

```diff
@@ -267,6 +269,28 @@ def reload(self) -> None:
         self.nics.clear()
         self._initialize()
 
+    @retry(tries=15, delay=3, backoff=1.15)
+    def wait_for_sriov_enabled(self) -> None:
+        lspci = self._node.tools[Lspci]
+
+        # check for VFs on the guest
+        vfs = lspci.get_devices_by_type(constants.DEVICE_TYPE_SRIOV)
+        assert_that(vfs).described_as(
+            "Could not identify any SRIOV NICs on the test node."
+        ).is_not_empty()
+
+        # check if the NIC driver has finished setting up the
+        # failsafe pair, reload if not
+        if not self.get_lower_nics():
+            self.reload()
+        if not self.get_lower_nics():
+            assert_that(self.get_lower_nics()).described_as(
+                "Did not detect any upper/lower sriov paired nics!: "
+                f"upper: {self.get_upper_nics()} "
+                f"lower: {self.get_lower_nics()} "
+                f"unpaired: {self.get_unpaired_devices()}"
+            ).is_not_empty()
```

Comment:
show we judge each synthetic has paired VF nic?

Reply:
The command is a lot of spew since we combined it into a single one to avoid roundtrips, but I'll add the list of found VF devices to this. The pairing is just from sysfs at /sys/class/net/[ethname]/lower_[lowerethname]/device

Reply:
check new code, is that good? Added a __str__ for PciDevice so we can print the list of VFs